### PR TITLE
Fixes Terra-UI Build issues and linter warnings

### PR DESCRIPTION
**Pull request by supreethmr**
### Summary

Fixes terra-ui build issue : https://github.com/cerner/terra-ui/actions/runs/4758810667/jobs/8457428665 

**What was changed:**

Change log title was corrected in terra-pills to match the existing change log formats and Added missing dependency of terra-worklist-data-grid is added 😆 

**Why it was changed:**

Because of Typo in terra-pills/CHANGELOG.md this file was not bundled with package and it was missing from node_modules/@cerner/terra-pills/ folder. Since change log was not part of component folder terra-ui build started to fail with module not found error while running `npm run build:production` command.

Screenshot 2023-06-08 at 7 34 23 PM

### Testing

Tested with tar.gz file terra-pills ( after typo correction ) in terra-ui to ensure node_modules/@cerner/terra-pills/CHANGELOG.md is available. 

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [X] Other (please describe below)
- [ ] No tests are needed

By running `npm run build:production` command on terra-ui 

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-XXXX 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by sdadn** [comment]
FYI, the terra-pills release should be fixed by https://github.com/cerner/terra-framework/pull/1666. We currently have https://github.com/cerner/terra-ui/pull/592 out to fix the build of terra-ui. The other changes in this PR look good though so we can keep it.

File: packages/terra-date-input/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * fixed linter warning
+
```
[the remark this answers is not in the document]
Comment (supreethmr) [reply packages/terra-date-input/CHANGELOG.md:7]: but 

> FYI, the terra-pills release should be fixed by #1666. We currently have [cerner/terra-ui#592](https://github.com/cerner/terra-ui/pull/592) out to fix the build of terra-ui. The other changes in this PR look good though so we can keep it.

is the build passing ?? and I don't think removing files from pills package.json will fix the build issue. have you tested terra-pill changes on terra-ui before merging PR #1666 ..??

**Comment by supreethmr** [comment]
> FYI, the terra-pills release should be fixed by #1666. We currently have [cerner/terra-ui#592](https://github.com/cerner/terra-ui/pull/592) out to fix the build of terra-ui. The other changes in this PR look good though so we can keep it.

is the build passing ?? and I don't think removing files from pills package.json will fix the build issue. have you tested terra-pill changes on terra-ui before merging PR #1666 ..??

File: packages/terra-date-input/CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (supreethmr) [reply packages/terra-date-input/CHANGELOG.md:7]: if we don't update changeLog will dangerJs not complain ??

**Comment by sdadn** [comment]
> is the build passing ?? and I don't think removing files from pills package.json will fix the build issue. was the code changes made on terra-pill tested on terra-ui before merging PR #1666 ..??

Yeah, here is a link to the build:
https://github.com/cerner/terra-ui/actions/runs/5223078737/jobs/9429612354

Here is a link to Terra UI with a new component that was introduced last week
https://engineering.cerner.com/terra-ui/components/cerner-terra-framework-docs/worklist-data-grid/about

The build failure was caused by the terra-pills CHANGELOG no longer being packaged upon release ([like here in v1.5.0](https://www.npmjs.com/package/@cerner/terra-pills/v/1.5.0?activeTab=code)) so #1666 fixes it with the [latest release](https://www.npmjs.com/package/@cerner/terra-pills/v/1.6.0?activeTab=code).

**Comment by supreethmr** [comment]
> > is the build passing ?? and I don't think removing files from pills package.json will fix the build issue. was the code changes made on terra-pill tested on terra-ui before merging PR #1666 ..??
> 
> Yeah, here is a link to the build: https://github.com/cerner/terra-ui/actions/runs/5223078737/jobs/9429612354
> 
> Here is a link to Terra UI with a new component that was introduced last week https://engineering.cerner.com/terra-ui/components/cerner-terra-framework-docs/worklist-data-grid/about
> 
> The build failure was caused by the terra-pills CHANGELOG no longer being packaged upon release ([like here in v1.5.0](https://www.npmjs.com/package/@cerner/terra-pills/v/1.5.0?activeTab=code)) so #1666 fixes it with the [latest release](https://www.npmjs.com/package/@cerner/terra-pills/v/1.6.0?activeTab=code).

Thank you for sharing the info will close this PR !!

**Comment by sdadn** [comment]
> Thank you for sharing the info will close this PR !!

We can still merge this PR for the linter fixes! Just need to remove the CHANGELOG entires.

**Comment by sdadn** [comment]
> Thank you for sharing the info will close this PR !!

@supreethmr We can still merge this PR for the linter fixes! Just need to remove the CHANGELOG entires.

**Pull request closed without merging**